src/Engine/Graphics/Direct3D/Render.cpp

```diff
@@ -1168,14 +1168,13 @@ void Render::DrawPolygon(struct Polygon *pPolygon) {
 
 Render::Render(
     std::shared_ptr<Application::GameConfig> config,
-    std::shared_ptr<OSWindow> window,
     DecalBuilder* decal_builder,
     LightmapBuilder* lightmap_builder,
     SpellFxRenderer* spellfx,
     std::shared_ptr<ParticleEngine> particle_engine,
     Vis* vis,
     Log* logger
-) : RenderBase(config, window, decal_builder, lightmap_builder, spellfx, particle_engine, vis, logger) {
```

Comment:
one thing we've been refactoring for the longest time is the plague of global variables .. what are the pros to have window,platform and others available globally?
i mean why wouldn't logger be available globally too, following this logic?

Reply:
This is temporary.

As I've noted above --- I needed to merge this in w/o rewriting everything, so I took some shortcuts.

Global variables will eventually be dealt with. Spelled this out explicitly in another TODO.